buildlib/cbuild

```diff
@@ -244,6 +244,13 @@ class travis(APTEnvironment):
         cmds.append("RUN " + " && ".join(scmds));
         return cmds;
 
+    def get_before_script(self):
+        """Return a list of commands to run from before_script"""
+        cmds = ["RUN useradd -ms /bin/bash travis"]
+        cmds.append("RUN cd /home/travis && mkdir build && " + " && ".join(self.yaml["before_script"]));
+        cmds.append("RUN chown travis /home/travis/build");
```

Comment:
Would this work?

```
cmds = ["RUN useradd -ms /bin/bash travis  && \\ "
        "su -l -c %s"%(pipes.quote(" && ".join(self.yaml["before_script"]))];
```
Eg run the before_script as the travis user as though run from a login shell